wrangler-transform/src/main/java/io/cdap/wrangler/Precondition.java

```diff
@@ -88,4 +98,32 @@ public boolean apply(Row row) throws PreconditionException {
       }
     }
   }
+
+  @Override
+  public Relation transform(RelationalTranformContext relationalTranformContext, Relation relation) {
+    if (!isConditionSQL) {
+      return new InvalidRelation("Condition is not specified as SQL");
+    }
+
+    Schema inputSchema = getSchema();
+
+    ExpressionConversionContext context = new ExpressionConversionContext();
```

Comment:
It's not clear where this would come from. It should come from the platform / app somehow, so it should be part of the relationalTranformContext. The main ask for this PR is to fully lay out the API / platform part, not the wrangler part